Creates the blog post for releasing Data Prepper 2.7.0 (#2654)
opensearch-project/project-website: _posts/2024-03-25-Announcing-Data-Prepper-2.7.0.md

---
layout: post
title:  "Announcing Data Prepper 2.7.0"
authors:
- dvenable
- qichen
date: 2024-03-25 13:30:00 -0600
categories:
  - releases
meta_keywords: GeoIP enrichment, AWS Secrets Manager Support, XML parsing, ION parsing, decompression
meta_description: Data Prepper 2.7.0 supports enriching events with GeoIP data, has support for using AWS Secrets Manager, and adds many new processors.
---

Data Prepper 2.7.0 is now available for [download](https://opensearch.org/downloads.html#data-prepper).
This release supports extracting geographic locations from IP addresses, supports injectable secrets, and adds many new processors.

## GeoIP processor

Data Prepper can now enrich events with geographical location data from an IP address using the new `geoip` processor.

> The `geoip` processor uses the MaxMind GeoLite2 databases to provide geographic location from IP addresses.

Reviewer: ```suggestion
The `geoip` processor uses the MaxMind GeoLite2 databases to provide geographical location data from IP addresses.
```

Many OpenSearch and Data Prepper users want to enrich their data by adding geographical locations to events.
There are a number of reasons this data can be valuable.

> Some examples are: looking for anomalies in network access; customer analytics; understanding load across geographies; and more.

Reviewer: ```suggestion
Some examples include customer analytics, looking for anomalies in network access, understanding load across geographies, and more.
```

An industry solution for determining a geographical location is through the use of IP addresses.

> One example scenario is locating users to a web server.

Reviewer: Should "to" be "of"?

Author: Yes, the word "of" is better here.

Data Prepper already supports parsing Apache Common Log Format for Apache HTTP servers in the `grok` processor.

> In the following example, we can now locate the client making requests using the `clientip` property extracted from the `grok` processor.

Reviewer: ```suggestion
The following example shows how you can now locate the client making requests using the `clientip` property extracted from the `grok` processor:
```

```
processor:
  - grok:
      match:
        log: [ "%{COMMONAPACHELOG_DATATYPED}" ]

  - geoip:
      entries:
        - source: clientip
          target: clientlocation
          include_fields: [latitude, longitude, location, postal_code, country_name, city_name]
```

When ingesting data using this pipeline, the OpenSearch index will now contain the geolocation fields expressed above, such as the `latitude` and `city_name`.
Additionally, you can configure template mappings in OpenSearch so that you can display these events in OpenSearch Dashboards using the [Maps feature](https://opensearch.org/docs/latest/dashboards/visualize/maps/).

## AWS Secrets Manager support

> Now Data Prepper supports [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/) as extension plugin applicable to pipeline plugins (i.e. source, buffer, processor, sink).

Reviewer: ```suggestion
Data Prepper now supports [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/) as an extension plugin applicable to pipeline plugins (source, buffer, processor, sink).
```

> Users are allowed to configure the AWS secrets extension through extensions in data-prepper-config.yaml.

Reviewer: Should "Secrets" be capitalized? Should the file name be in code font?

Author: I changed to "AWS Secrets Manager"

> Here is an example of how to configure your secrets.

Reviewer: ```suggestion
The following example shows how you can configure your secrets:
```

```
extensions:
  aws:
    secrets:
      host-secret-config:
        secret_id: <YOUR_SECRET_ID_1>
        region: <YOUR_REGION_1>
        sts_role_arn: <YOUR_STS_ROLE_ARN_1>
        refresh_interval: <YOUR_REFRESH_INTERVAL_1>
      credential-secret-config:
        secret_id: <YOUR_SECRET_ID_2>
        region: <YOUR_REGION_2>
        sts_role_arn: <YOUR_STS_ROLE_ARN_2>
        refresh_interval: <YOUR_REFRESH_INTERVAL_2>
```

> Users can also configure this in the `pipeline_configurations` section of a pipeline YAML file.

Reviewer: Instead of "this", let's name what we're referencing. "secrets"?

Author: I changed "this" to "secrets."

> The `credential-secret-config` is a user-supplied secret configuration Id.

Reviewer: ```suggestion
The `credential-secret-config` is a user-supplied secret configuration ID.
```

> The `credential-secret-config` is a user-supplied secret configuration Id.

Reviewer: A noun should follow the text in code font.

Author: Please see my update and let me know if you have other suggestions.

Reviewer: Works for me 😄

> Pipeline authors can reference secrets within pipeline plugin settings using the pattern of `$aws_secrets:<<my-defined-secret>>``.

Reviewer: ```suggestion
Pipeline authors can reference secrets within pipeline plugin settings using the pattern `$aws_secrets:<<my-defined-secret>>``.
```

> Here is an example of configuring an OpenSearch sink with secret values.

Reviewer: ```suggestion
The following example shows how to configure an OpenSearch sink with secret values:
```

```
{% raw %}
source:
  - opensearch:
      hosts: [ "${{aws_secrets:host-secret-config}}" ]
      username: "${{aws_secrets:credential-secret-config:username}}"
      password: "${{aws_secrets:credential-secret-config:password}}"
{% endraw %}
```

> In the above example, secrets under `credential-secret-config` is assumed to be stored as the following JSON key-value pairs:

Reviewer: ```suggestion
In this example, secrets under `credential-secret-config` are assumed to be stored as the following JSON key-value pairs:
```

```
{
  "username": <YOUR_USERNAME>
  "password": <YOUR_PASSWORD>
}
```

The secret under `host-secret-config` is assumed to be stored as plaintext.

> To support [secret rotation](https://docs.aws.amazon.com/secretsmanager/latest/userguide/rotating-secrets.html) for OpenSearch, the `opensearch` source automatically refreshes its basic credentials, (i.e. username/password) according to the `refresh_interval` by polling the latest secret values.

Reviewer: ```suggestion
To support [secret rotation](https://docs.aws.amazon.com/secretsmanager/latest/userguide/rotating-secrets.html) for OpenSearch, the `opensearch` source automatically refreshes its basic credentials, (username/password) according to the `refresh_interval` by polling the latest secret values.
```

> For more details, please reference our [aws extension plugins](https://opensearch.org/docs/latest/data-prepper/managing-data-prepper/configuring-data-prepper/#aws-extension-plugins) document.

Reviewer: ```suggestion
For more information, please see the [AWS extension plugins](https://opensearch.org/docs/latest/data-prepper/managing-data-prepper/configuring-data-prepper/#aws-extension-plugins) documentation.
```

Author: We actually call this extension `aws`. Perhaps we can change this to:

For more information, please see the [`aws` extension](https://opensearch.org/docs/latest/data-prepper/managing-data-prepper/configuring-data-prepper/#aws-extension-plugins) documentation.

Reviewer: I think the link is referencing the heading in the documentation.

Author: We are calling these "extensions" in Data Prepper, not "extension plugins." The "plugin" part is redundant in both.

## Other features

* Data Prepper can now parse XML data in fields using the `parse_xml` processor.

> * The new `parse_ion` processor can parse fields which are in the [Amazon Ion format](https://amazon-ion.github.io/ion-docs/).

Reviewer: ```suggestion
* The new `parse_ion` processor can parse fields in the [Amazon Ion format](https://amazon-ion.github.io/ion-docs/).
```

> * Some users have fields which are gzip-compressed at the field level. These users can decompress that data using the `decompress` processor.

Reviewer: ```suggestion
* Some users have fields that are gzip-compressed at the field level. These users can decompress that data using the `decompress` processor.
```

> * Some users have fields which are gzip-compressed at the field level. These users can decompress that data using the `decompress` processor.

Reviewer: The noun following "that" in the second sentence should match the noun in the first sentence (fields/those fields or data that is/that data).

> * Data Prepper has a new function which can join strings from multiple strings together, including with a delimiter.

Reviewer: ```suggestion
* Data Prepper can now join strings from multiple strings, including with a delimiter.
```

> * The new `select_entries` processor allows users to select only the fields that they need from their events. This can simplify how users filter out unnecessary data.

Reviewer: ```suggestion
* The new `select_entries` processor allows users to select only the necessary fields from events. This can simplify how users filter unnecessary data.
```

* The `file` source now supports codecs. This can help you test a pipeline locally before using the `s3` source.

## Getting started

* To download Data Prepper, see the [OpenSearch downloads](https://opensearch.org/downloads.html) page.
* For instructions on how to get started with Data Prepper, see [Getting started with Data Prepper](https://opensearch.org/docs/latest/data-prepper/getting-started/).
* To learn more about the work in progress for Data Prepper 2.8 and other releases, see the [Data Prepper roadmap](https://github.com/opensearch-project/data-prepper/projects/1).

## Thanks to our contributors!

> The following people contributed to this release. Thank you!

Reviewer: Ensure this is added.
